src/main/sensors/gyro.c

```diff
@@ -300,6 +330,22 @@ void gyroUpdate(void)
         }
     }
 
+#ifdef USE_GYRO_NOTCH_1
+    if (gyroConfig->gyro_soft_notch_hz_1){
+        for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++){
+            gyro.gyroADC[axis] = lrintf(biquadFilterApply(&gyroFilterNotch1[axis], (float) gyro.gyroADC[axis]));
+        }	
+    }
+#endif
+
+#ifdef USE_GYRO_NOTCH_2
+    if (gyroConfig->gyro_soft_notch_hz_2){
+        for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++){
+            gyro.gyroADC[axis] = lrintf(biquadFilterApply(&gyroFilterNotch2[axis], (float) gyro.gyroADC[axis]));
+        }	
+    }
+#endif
+	
```

Comment:
Have a look at the betaflight code. By setting up function pointers to the filters this code can be made neater and more efficient. Eg:
```
    float gyroADCf[3];
    for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
        gyroADCf[axis] = softLpfFilterApplyFn(softLpfFilter[axis], (float)gyro.gyroADC[axis]));
        gyroADCf[axis] = notchFilter1ApplyFn(notchFilter1[axis], gyroADCf[axis]);
        gyroADCf[axis] = notchFilter2ApplyFn(notchFilter2[axis], gyroADCf[axis]);
        gyro.gyroADC[axis] = lrintf(gyroADCf[axis]);
    }
```
This removes the intermediate calls to `lrintf` which are both expensive and add rounding errors. It also avoids the `if (gyroConfig->gyro_soft_notch_hz_N)` tests which are ungainly.